src/scp/test/SCPTests.cpp

```diff
@@ -2821,6 +2821,48 @@ TEST_CASE("nomination tests core5", "[scp][nominationprotocol]")
                 verifyNominate(scp.mEnvs[1], v0SecretKey, qSetHash0, 0, votesXK,
                                {});
             }
+            SECTION("select accepted value from leader")
+            {
+                REQUIRE(xValue < yValue);
+                REQUIRE(yValue < zValue);
+                REQUIRE(scp.mEnvs.size() == 1);
+
+                // Update round leader to v1
+                scp.mPriorityLookup = [&](NodeID const& n) {
+                    return (n == v1NodeID) ? 1000 : 1;
+                };
+
+                SCPEnvelope nom1 = makeNominate(v1SecretKey, qSetHash, 0,
+                                                {yValue, zValue}, {yValue});
+
+                SECTION("receive accepted before timeout")
+                {
+                    // nothing more happens, v0 is leader
+                    scp.receiveEnvelope(nom1);
+                    REQUIRE(scp.mEnvs.size() == 1);
+
+                    // Update round leaders, vote for accepted value (z)
+                    REQUIRE(scp.nominate(0, xValue, true));
+                    REQUIRE(scp.mEnvs.size() == 2);
+                }
+                SECTION("receive accepted after timeout")
+                {
+                    REQUIRE(!scp.nominate(0, xValue, true));
+                    REQUIRE(scp.mEnvs.size() == 1);
+
+                    // Vote for accepted value (z)
+                    scp.receiveEnvelope(nom1);
+                    REQUIRE(scp.mEnvs.size() == 2);
+                }
+
+                std::vector<Value> votesXY;
+                votesXY.emplace_back(xValue);
+                votesXY.emplace_back(yValue);
+                std::sort(votesXY.begin(), votesXY.end());
+
+                verifyNominate(scp.mEnvs[1], v0SecretKey, qSetHash0, 0, votesXY,
```

Comment:
you're missing receiving
`SCPEnvelope nom2 = makeNominate(v1SecretKey, qSetHash, 0,
                                                {yValue, zValue, zzValue}, {yValue});`
and perform checks that if we receive this before and after timeout we still only vote for the same thing (this would have caught the bug that you had in your first implementation)

Reply:
I can add this, though I don't think both before/after timeout sections are needed. We already tested above that the updated function is called appropriately in different scenarios wrt timeout. Here we just need to test that the actual function does what we want (i.e. does not select votes)

Reply:
agree - we just need to test any code path that hits `getNewValueFromNomination`